pigeon/weights/models.py

```diff
@@ -133,6 +140,24 @@ def update_with_OFF_product(self, product_off):
             if off_field != getattr(self, field):
                 setattr(self, field, off_field)
 
+    @property
+    def quantity_grams(self):
+        qte = self.quantity.lower()
+        qte = re.sub(r"\s+", "", qte)
+        kg_pat = re.match(r'^([\d.,]+)kg$', qte)
+        if kg_pat:
+            return float(kg_pat.groups()[0]) * CONVERSIONS['kg__g']
+        oz_pat = re.match(r'^([\d.,]+)oz$', qte)
+        if oz_pat:
+            return float(oz_pat.groups()[0]) * CONVERSIONS['oz__g']
+        lb_pat = re.match(r'^([\d.,]+)lbs?$', qte)
+        if lb_pat:
+            return float(lb_pat.groups()[0]) * CONVERSIONS['lb__g']
+        g_pat = re.match(r'^([\d.,]+)(g|gr|grammes|gramm|gram|grams)$', qte)
+        if g_pat:
+            return float(g_pat.groups()[0])
+        return ""
+
```

Comment:
Maybe make a private func to make all the conversions.
All the returns are unreadables, move that into something nobody will read ;-)